## Waze's Predict User Churn Project

**Stats and Hypothesis Testing**

$Task:$ To find out whether there is a statistically significant difference in mean amount of rides between the 2 device type users.

*1. Importing and Data Loading*

In [1]:
#Import libraries or packages
import os

#Data Manipulation
import pandas as pd
import numpy as np

#Data Visualization
import matplotlib.pyplot as plt
import seaborn as sns

#Statistics
from scipy import stats

In [2]:
#Import the dataset
get_cwd = os.getcwd()
df = pd.read_csv(get_cwd + '/waze_dataset.csv')

*2. Data Exploration*

In [3]:
#1. Create `map_dictionary`
map_dictionary = {'Android': 2, 'iPhone': 1}

#2. Create new `device_type` column and map the new column to the dictionary
df['device_type'] = df['device'].map(map_dictionary)
df.head(10)

,ID,label,sessions,drives,total_sessions,n_days_after_onboarding,total_navigations_fav1,total_navigations_fav2,driven_km_drives,duration_minutes_drives,activity_days,driving_days,device,device_type
0,0,retained,283,226,296.748273,2276,208,0,2628.845068,1985.775061,28,19,Android,2
1,1,retained,133,107,326.896596,1225,19,64,13715.920550,3160.472914,13,11,iPhone,1
2,2,retained,114,95,135.522926,2651,0,0,3059.148818,1610.735904,14,8,Android,2
3,3,retained,49,40,67.589221,15,322,7,913.591123,587.196542,7,3,iPhone,1
4,4,retained,84,68,168.247020,1562,166,5,3950.202008,1219.555924,27,18,Android,2
5,5,retained,113,103,279.544437,2637,0,0,901.238699,439.101397,15,11,iPhone,1
6,6,retained,3,2,236.725314,360,185,18,5249.172828,726.577205,28,23,iPhone,1
7,7,retained,39,35,176.072845,2999,0,0,7892.052468,2466.981741,22,20,iPhone,1
8,8,retained,57,46,183.532018,424,0,26,2651.709764,1594.342984,25,20,Android,2
9,9,churned,84,68,244.802115,2997,72,0,6043.460295,2341.838528,7,3,iPhone,1


In [4]:
#Calculate the mean of the number of drives for each device type
df.groupby('device_type')['drives'].mean()

device_type
1    67.859078
2    66.231838
Name: drives, dtype: float64

*3. Hypothesis Testing*

Steps:
1.   State the null hypothesis and the alternative hypothesis
2.   Choose a signficance level
3.   Find the p-value
4.   Reject or fail to reject the null hypothesis

*Step 1:*

Null Hypothesis(H0): There is no difference in avg between drives who use an iPhone and drivers who use an Android.

Alternative Hypothesis(H1): There is a difference in avg between drives who use an iPhone and drivers who use an Android.

*Step 2:*

Significane Level: 5% or 0.05

In [5]:
#Significance Level
sl = 0.05

*Step 3: Finding the P-Value* 

In [6]:
#1. Isolate the `drives` column for iPhone users.
iPhone = df[df['device_type'] == 1]['drives']

#2. Isolate the `drives` column for Android users.
Android = df[df['device_type'] == 2]['drives']

In [7]:
#3. Perform the t-test
result = stats.ttest_ind(a = iPhone, b = Android, equal_var=False)
p_value = result[1]
print("P-value:", p_value*100)
print("P-value < significance level (5%):", p_value < sl)

P-value: 14.3351972680206
P-value < significance level (5%): False


*Step 4: Reject or Fail to Reject the Null Hypothesis*

Since the P-value is 14.3%, i.e, more than the confidence level, we fail to reject the null hypothesis. You conclude that there is **not** a statistically significant difference in the average number of drives between drivers who use iPhones and drivers who use Androids.

**Conclusion:**

The key business insight is that drivers who use iPhone devices on average have a similar number of drives as those who use Androids.

One potential next step is to explore what other factors influence the variation in the number of drives, and run additonal hypothesis tests to learn more about user behavior. Further, temporary changes in marketing or user interface for the Waze app may provide more data to investigate churn.